projects/types.nix

```diff
@@ -7,8 +7,30 @@ let
     types
     mkOption
     ;
+
+  types' = import ./types.nix { inherit lib; };
```

Comment:
Does this make sense here to distinguish Nixpkgs `types` from our custom types or should we just use `rec`?

Reply:
It's a rather weird thing to import the same file, this tangibly increases WTFs/min while reading. `let types' = {}; in types'` could work, but If I'm not overlooking something it's equivalent to using `rec` (modulo some edge cases where one or the other is slightly easier to read).

Reply:
It's kind of odd for sure, but I'm wondering if it would be more readable if custom types were referenced by name. It would be interesting to see a namespaced `rec` for something like this. I'm not sure about this, myself, so feel free to remove this if you'd like.